internal/datacoord/compaction_trigger.go

```diff
@@ -285,6 +286,10 @@ func (t *compactionTrigger) triggerSingleCompaction(collectionID, partitionID, s
 // forceTriggerCompaction force to start a compaction
 // invoked by user `ManualCompaction` operation
 func (t *compactionTrigger) forceTriggerCompaction(collectionID int64) (UniqueID, error) {
+	if !compactionEnabled() {
```

Comment:
According to the change, if the compaction is disabled, even the manual compaction get disabled, too. This may unnecessarily violate backward compatibility. 

Also, this makes `forceTriggerCompaction` not “force” at all, which is confusing.

Reply:
![image](https://github.com/milvus-io/milvus/assets/51370125/417182b2-6fde-4bab-bf98-67d1223bf72f) EnableCompaction config controls ManualCompaction by design, This pr just move the check inside compaction_trigger so that caller don't have to concern about availability of compaction_trigger.  

Remove it would be violate backward compactibility.